instrumentation/instagin/handler_test.go

```diff
@@ -0,0 +1,116 @@
+// (c) Copyright IBM Corp. 2021
+// (c) Copyright Instana Inc. 2016
+
+// +build go1.11
+
+package instagin_test
+
+import (
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/instana/testify/require"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/instana/testify/assert"
+
+	"github.com/gin-gonic/gin"
+	instana "github.com/instana/go-sensor"
+	"github.com/instana/go-sensor/instrumentation/instagin"
+)
+
+func TestAddMiddleware(t *testing.T) {
```

Comment:
Is this test still needed after we removed the `findAndReplace()`? I'd rather suggest that we check if an instrumented handler collects expected information in a span, i.e. the path template, URL, params, etc.

Reply:
This test, checks that middleware added as a first handler, no matter if there are some middleware already added. I will add a test to check data, as you suggested.

Reply:
I see 👍  I'm fine with current implementation, but as a note you might have added a middleware and check that there is a span in request context there, since this is wha we're trying to achieve. This way we're not dependent on Gin implementation

Reply:
done.